Applications/yOCTScanAndFindTissueSurface.m

```diff
@@ -1,44 +1,143 @@
+addParameter(p,'v',false);
+addParameter(p, 'acceptableRange_mm', 0.025, @isnumeric)
 
 parse(p,varargin{:});
 in = p.Results;
 xRange_mm = in.xRange_mm;
 yRange_mm = in.yRange_mm;
-resolution_mm = in.resolution_mm;
+pixel_size_um = in.pixel_size_um;
 isVisualize = in.isVisualize;
+octProbePath = in.octProbePath;
+output_folder = in.output_folder;
+dispersionQuadraticTerm = in.dispersionQuadraticTerm;
+acceptableRange_mm = in.acceptableRange_mm;
+v = in.v;
 
 %% Scan
-% Todo
+totalStartTime = datetime;  % Capture the starting time
+volumeOutputFolder = [output_folder '/OCTVolume/'];
+disp('Please adjust the OCT focus such that it is precisely at the intersection of the tissue and the coverslip.')
+fprintf('%s Scanning Volume...\n', datestr(datetime));
+scanParameters = yOCTScanTile (...
+    volumeOutputFolder, ...
+    xRange_mm, ...
+    yRange_mm, ...
+    'octProbePath', octProbePath, ...
+    'pixelSize_um', pixel_size_um, ...
+    'v',v  ...
+    );
+
+%% Check if focusPositionInImageZpix is provided, if not use yOCTFindFocusTilledScan
+if isempty(in.focusPositionInImageZpix) || isnan(in.focusPositionInImageZpix)
+    fprintf('%s Find focus position volume\n', datestr(datetime));
+    focusPositionInImageZpix = yOCTFindFocusTilledScan(volumeOutputFolder,...
+        'reconstructConfig', {'dispersionQuadraticTerm', dispersionQuadraticTerm}, 'verbose', true);
+else
+    focusPositionInImageZpix = in.focusPositionInImageZpix;
+end
+
+%% Reconstruct OCT Image for Subsequent Surface Analysis
+fprintf('Generating image...\n');
+outputTiffFile = [output_folder '\surface_analysis.tiff'];
+yOCTProcessTiledScan(...
+    volumeOutputFolder, ... Input
+    {outputTiffFile},... Save only Tiff file as folder will be generated after smoothing
+    'focusPositionInImageZpix',focusPositionInImageZpix,...
+    'dispersionQuadraticTerm',dispersionQuadraticTerm,...
+    'cropZAroundFocusArea', false,...
+    'v',v);
+[logMeanAbs, dimensions, ~] = yOCTFromTif(outputTiffFile);
+if (v)
+    disp([newline, '-- Data loaded successfully.']);
+end
 
 %% Estimate tissue surface
-% Todo
+if (v)
+    fprintf('%s Identifying tissue surface...\n', datestr(datetime));
+end
+tic;
+dimensions = yOCTChangeDimensionsStructureUnits(dimensions,'millimeters'); % Make sure it's in mm
+surfacePosition_mm = yOCTFindTissueSurface(logMeanAbs, dimensions, 'isVisualize', isVisualize);
+x_mm=dimensions.x.values;
+y_mm=dimensions.y.values;
+elapsedTimeSurfaceDetection = toc;
+if (v)
+    fprintf('%s Surface identification completed in %.2f seconds.\n', datestr(datetime), elapsedTimeSurfaceDetection);
+end
 
 %% Clean up
-% delete temporary files
+if exist(output_folder, 'dir')
+    rmdir(output_folder, 's'); % Remove the output directory after processing
+end
+totalEndTime = datetime;  % Capture the ending time
+totalDuration = totalEndTime - totalStartTime;
+if (v)
+    fprintf('Total surface analysis process completed in %s.\n', datestr(totalDuration, 'HH:MM:SS'));
```

Comment:
Format
 fprintf('%s yOCTScanAndFindTissueSurface function evaluation completed in %s.\n', datestr(datetime),
datestr(totalDuration, 'HH:MM:SS'));